app/web/features/search/SearchPage.tsx

```diff
@@ -168,6 +166,31 @@ export default function SearchPage({
 
   const errorMessage = error?.message;
 
+  const handleHostingStatusFilterChange = (
+    event: ChangeEvent<{ value: unknown }>
+  ) => {
+    console.log("EVENT VALUE", event.target.value);
+    const value = event.target.value as Exclude<
+      HostingStatus,
+      HostingStatus.HOSTING_STATUS_UNKNOWN
+    >; // Type assertion to HostingStatus array
+    if (hostingStatusFilter === undefined) {
+      setHostingStatusFilter([value]);
+    } else if (hostingStatusFilter.includes(value)) {
+      const newHostingStatusFilter = hostingStatusFilter.filter(
+        (existingValue) => existingValue !== value
+      );
+
+      if (hostingStatusFilter.length === 0) {
+        setHostingStatusFilter(undefined);
+      } else {
+        setHostingStatusFilter(newHostingStatusFilter);
+      }
+    } else {
+      setHostingStatusFilter([...hostingStatusFilter, value]);
+    }
+  };
```

Comment:
In plain language, when the user chooses a value, since it's a multiselect we want to:

```javascript
    if (hostingStatusFilter === undefined) { // if no values yet, add the first one in an array
      setHostingStatusFilter([value]);
    } else if (hostingStatusFilter.includes(value)) { // if the value is already there, remove it
      const newHostingStatusFilter = hostingStatusFilter.filter(
        (existingValue) => existingValue !== value
      );

      if (hostingStatusFilter.length === 0) { // if removing the last value, set to undefined again
        setHostingStatusFilter(undefined);
      } else {
        setHostingStatusFilter(newHostingStatusFilter);
      }
    } else {
      setHostingStatusFilter([...hostingStatusFilter, value]); // otherwise, just add it to existing values
    }
  };
```